service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/utils/DocumentReferenceUtils.java

```diff
@@ -0,0 +1,36 @@
+package uk.nhs.adaptors.gp2gp.ehr.utils;
+
+import org.hl7.fhir.dstu3.model.Attachment;
+import org.hl7.fhir.dstu3.model.DocumentReference;
+import uk.nhs.adaptors.gp2gp.ehr.exception.EhrMapperException;
+
+import java.util.Optional;
+
+public class DocumentReferenceUtils {
+    public static Attachment extractAttachment(DocumentReference documentReference) {
+        return documentReference.getContent()
+            .stream()
+            .findFirst()
+            .filter(DocumentReference.DocumentReferenceContentComponent::hasAttachment)
+            .map(DocumentReference.DocumentReferenceContentComponent::getAttachment)
+            .orElseThrow(() -> new EhrMapperException("documentReference.content[0] is missing an attachment"));
+    }
+
+    public static String extractContentType(Attachment attachment) {
+        return Optional.ofNullable(attachment.getContentType())
+            .orElseThrow(() -> new EhrMapperException("documentReference.content[0].attachment is missing contentType"));
+    }
+
+    public static String buildAttachmentFileName(String narrativeStatementId, Attachment attachment) {
+        var fileExtension = mapContentTypeToFileExtension(extractContentType(attachment));
+
+        return Optional.ofNullable(attachment.getTitle())
+            .map(__ -> "AbsentAttachment" + narrativeStatementId + fileExtension)
```

Comment:
If the attachment has a title, map it to "AbsentAttachment"? It's not obvious to me why that would be the case.

Reply:
It's not my code. I talked with @n-r-b about that as she implemented this in her https://github.com/nhsconnect/integration-adaptor-gp2gp/pull/146 and there was some discussion about this. Apparently this is the expected behaviour